src/code/TwoHeadArena.c

```diff
@@ -32,7 +32,7 @@ void* THA_GetTail(TwoHeadArena* tha) {
 void* THA_AllocHead(TwoHeadArena* tha, size_t size) {
     void* start = tha->head;
 
-    tha->head = (u8*)tha->head + size;
+    tha->head = (void*)((uintptr_t)tha->head + size);
```

Comment:
I don't really see the point of casting out of and back into a pointer here when it's not necessary, (u8*) just seems better for this sort of arithmetic where possible.

Reply:
Personally I see doing pointer arithmetic as something bad and which should be avoided. That's why we have types to do proper pointer arithmetic

Reply:
But this is standard pointer arithmetic, not alignment, it's not an unusual operation. Arithmetic is one of the two main properties of pointers, after all.

Reply:
Casting to `uintptr_t` to perform pointer arithmetic is way more common in the codebase than using `u8*`